Use datetime objects everywhere

mattoufoutu opened the pull request:
This pull request implements datetime objects in every fields where a date is expected, allowing the values to be comparable in any context (be it in Python or directly in a Mongo query).

PidgeyL commented:
I assume this would require a database drop and repopulate, after implementing?

mattoufoutu commented:
Yes it would, unfortunately.

PidgeyL commented:
Okay. Does the date stay in UTC, and does the web interface still translates this to the current time zone?

mattoufoutu commented:
The date stays in UTC, but I didn't take the time to test the web interface (it's a part I don't use).
If it breaks things, just let me know, I'll add the required fixes to the PR.

PidgeyL commented:
I will test this then. @wimremes I will test this either today or tomorrow, I will tell you the results

PidgeyL commented:
@mattoufoutu it indeed broke the web component. Could you please fix this?

mattoufoutu commented:
No problem, I'll have a look at it during the week-end. I'll update the PR as soon as it's ok.

PidgeyL commented:
Okay, sounds good :)

adulau commented:
If there is a branch on your repo to test before @mattoufoutu feel free to let us know. Thank you.

mattoufoutu commented:
Updated dates display (new implementation allows to specify a different format if needed by passing it to the filter: `var|currentTime('%d/%m/%Y %H:%M:%S' )`)
Also cleaned up some code (no big changes).

I spotted multiple variables that are defined but never used, but I left them in case it's intended for a future piece of code or whatever.

Does it seems ok to you?

@adulau: I forgot to create a different branch before starting my changes, but you can still test against my master branch.

PidgeyL commented:
@mattoufoutu Thanks for the contributions, it's much appreciated

mattoufoutu commented:
the search.py script was broken because of my changes, latest commit fixes it

mattoufoutu commented:
I updated my PR to be merge-able with current master branch

jvoisin commented:
Can we get this merged?

PidgeyL commented:
I pulled this in my repo. Changes will be pulled to the main repo as soon as @adulau merges
